src/app/ipv/middleware.js

```diff
@@ -131,7 +131,7 @@ async function handleBackendResponse(req, res, backendResponse) {
 
     req.session.ipvSessionId = null;
     const { redirectUrl } = backendResponse.client;
-    return res.redirect(redirectUrl);
+    return saveSessionAndRedirect(req, res, redirectUrl);
```

Comment:
Not at all needed right now, but do we need something similar on line 107?

Reply:
Yeah potentially - it's even less likely to be an issue when redirecting to a CRI (since they shouldn't immediately bounce back), but would be good to be consistent.